apps/angular/src/app/pages/main-page.component.ts

```diff
@@ -0,0 +1,152 @@
+	styleUrl: './main-page.component.css',
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MainPageComponent implements OnInit {
+	private readonly animeService = inject(AnimeService);
+
+	private readonly activatedRoute = inject(ActivatedRoute);
+
+	private readonly navigationService = inject(NavigationService);
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+		this.subscribeToQueryParams();
+	}
+
+	/**
+	 * Update page number and page size.
+	 * @param event Page event.
+	 */
+	protected onPageEvent(event: PageEvent): void {
+		this.pageNumber$.next(event.pageIndex);
+		this.pageSize$.next(event.pageSize);
+	}
+
+	/**
+	 * Handle sort change.
+	 * @param event Sort change event.
+	 */
+	protected onSortChange(event: Sort): void {
+		this.ordering$.next({ activeSort: event.active as SortParams, direction: event.direction });
+	}
+
+	/** Handle search value change.
+	 * @param event Search value.
+	 */
+	protected onSearchValueChange(event: string): void {
+		this.search$.next(event);
+		this.pageNumber$.next(DEFAULT_PAGE_NUMBER);
+	}
+
+	/**
+	 * Handle filter value change.
+	 * @param event Type filter change event.
+	 */
+	protected onFilterValueChange(event: readonly AnimeType[]): void {
+		this.animeTypeFilter$.next(event);
+		this.pageNumber$.next(DEFAULT_PAGE_NUMBER);
+	}
+
+	private getAnimeList(params: AnimeManagementParams): Observable<Pagination<Anime>> {
+		return this.animeService.getPaginatedAnime(params);
+	}
+
+	private navigateToRouteWithParams(params: AnimeManagementParams): void {
+		this.navigationService.navigate('', AnimeManagementParamsMapper.toQueryParams(params));
+	}
+
+	private subscribeToQueryParams(): void {
+		this.activatedRoute.queryParams
+			.pipe(
+				map(params => AnimeManagementParamsMapper.fromQueryParams(params as AnimeManagementParamsDto)),
+				tap(params => {
+					this.pageSize$.next(params.pageSize);
+					this.pageNumber$.next(params.pageNumber);
+					this.search$.next(params.search ?? '');
+					this.ordering$.next(params.ordering != null ? SortMapper.fromDto(params.ordering) : null);
+					this.animeTypeFilter$.next(params.types);
+				}),
+				takeUntilDestroyed(this.destroyRef),
+			)
+			.subscribe();
+	}
+
+	private createAnimeListStream(): Observable<Pagination<Anime>> {
+		return combineLatest([this.pageNumber$, this.pageSize$, this.ordering$, this.search$, this.animeTypeFilter$]).pipe(
+			debounceTime(DEBOUNCE_TIME),
+			map(([pageNumber, pageSize, ordering, search, filter]) => ({
+				pageSize,
+				pageNumber,
+				ordering: ordering != null ? SortMapper.toDto(ordering) : undefined,
+				search: search.length > 0 ? search : undefined,
+				types: filter,
+			})),
+			tap(params => this.navigateToRouteWithParams(params)),
+			switchMap(params => this.getAnimeList(params)),
+			catchError((error: unknown) => {
+				this.hasFetchingError$.next(true);
+				return throwError(() => error);
+			}),
```

Comment:
Throwing error further might be reasonable in API service. However, it's not a good idea to do so in a component (i.e. UI layer).
If we don't handle this error completely it might break our application (if error boundary is not configured correctly).

In your case you already handle the error - the error is displayed to user in the UI. Since we handle the error there is no need to throw error further. It's common to return `EMPTY` (from RxJS) after error is handled:
```suggestion
			catchError((error: unknown) => {
				this.hasFetchingError$.next(true);
				return EMPTY;
			}),
```